src/core/util/constant.js

```diff
@@ -7,9 +7,22 @@ angular.module('material.core')
  */
 function MdConstantFactory($sniffer) {
 
-  var webkit = /webkit/i.test($sniffer.vendorPrefix);
+  var isWebkit = /webkit/i.test($sniffer.vendorPrefix);
+  var prefixTestEl = document.createElement('div');
```

Comment:
To be honest, this code is much more complicated as the code before. 

- It uses a RegEx specially for the `Moz` prefix.
- Using a RegEx and iterate through all groups is unnecessary, because we only need to modify the first letter.

I like the `hasStyle` function, this is very clean.
I also like the idea of the `camelCase` function, but it can be improved, just for our cases.

````js
function camelCase(input, isLowerCase) {
 return (isLowerCase() ? name.charAt(0).toLowerCase() : 
                         name.charAt(0).toUpperCase()) + name.substring(1);
}
```